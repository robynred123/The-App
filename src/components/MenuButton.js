import React from 'react'
import { StyleSheet, Text, View, TouchableOpacity } from 'react-native'
/*
can add in on ejecting, package not accepted by Expo
prevents passing in navigation per button
import { useNavigation } from '@react-navigation/native';
*/

export const MenuButton = (props) => {
  //const navigation = useNavigation()
  const { title, navigateTo, navigation } = props
  return (
    <View style={styles.container}>
      <TouchableOpacity
        onPress={() => navigation.navigate(navigateTo)}
        style={styles.button}
      >
        <Text style={styles.buttonText}>{title}</Text>
      </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  container: {
    justifyContent: 'space-around',
    padding: 10
  },
  button: {
    backgroundColor: '#AA7DC3',
    justifyContent: 'center',
    alignItems: 'center',
    padding: 15,
    borderRadius: 15
  },
  buttonText: {
    fontSize: 15,
    fontWeight: '700'
  }
})
