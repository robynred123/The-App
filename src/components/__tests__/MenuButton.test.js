import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { MenuButton } from '../MenuButton';

Enzyme.configure({ adapter: new Adapter() })

const navigation = { navigate: jest.fn() };

const onPress = jest.fn()

describe('<MenuButton />', () => {
    const component = shallow(<MenuButton title='test' navigateTo='test' navigation={navigation}/>)

    it('should render the MenuButton without crashing', () => {
        expect(component).toMatchSnapshot()
    })

    it('should render the MenuButton with a title of "test"', () => {
        expect(component.find('Text').props().children).toEqual('test')
    })

   /* it('should render the MenuButton and navigate on press', () => {
        //Enzyme adapter returning ForwardRef instead of TouchableOpacity
        const button = component.find('ForwardRef').props().onPress
        
        expect(onPress).toHaveBeenCalledTimes(1)
    })*/
})