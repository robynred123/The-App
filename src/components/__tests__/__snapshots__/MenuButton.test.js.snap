// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuButton /> should render the MenuButton without crashing 1`] = `
<View
  style={
    Object {
      "justifyContent": "space-around",
      "padding": 10,
    }
  }
>
  <ForwardRef
    onPress={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#b067d8",
        "borderRadius": 15,
        "justifyContent": "center",
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 15,
          "fontWeight": "700",
        }
      }
    >
      test
    </Text>
  </ForwardRef>
</View>
`;
